yarn-project/simulator/src/avm/avm_execution_environment.ts

```diff
@@ -22,29 +22,18 @@ export class AvmContextInputs {
 export class AvmExecutionEnvironment {
   constructor(
     public readonly address: AztecAddress,
-
     public readonly storageAddress: AztecAddress,
-
     public readonly origin: AztecAddress,
-
     public readonly sender: AztecAddress,
-
     public readonly portal: EthAddress,
-
     public readonly feePerL1Gas: Fr,
-
     public readonly feePerL2Gas: Fr,
-
     public readonly feePerDaGas: Fr,
-
     public readonly contractCallDepth: Fr,
-
+    public readonly header: Header,
```

Comment:
I think we might eventually want opcodes to directly get certain entries from this header (different from HEADERMEMBER which gets historic info)

Reply:
Yes, we can do all of it from DB at that point. This was the easy way out for now :)
BTW, when I discussed with Lasse, he thought that HEADERMEMBER only made sense w/the latest header, because all the rest you can do in private.